modules/version-checker/version_checker.go

```diff
@@ -0,0 +1,190 @@
+	// Check for empty parameters
+	if len(params.expectedVersion) == 0 {
+		return fmt.Errorf("empty expectedVersion found")
+	} else if len(params.workingDir) == 0 {
+		return fmt.Errorf("empty workingDir found")
+	} else if params.binary < 0 {
+		return fmt.Errorf("empty binary found")
+	}
+
+	// Check the format of the expected version.
+	match, err := regexp.MatchString(VersionRegexMatcher, params.expectedVersion)
+	if err != nil {
+		return fmt.Errorf(
+			"unexpected regex matcher compilation failure: %w", err)
+	} else if !match {
+		return fmt.Errorf(
+			"invalid expected version format found {%s}", params.expectedVersion)
+	}
+
+	return nil
+}
+
+// getVersionWithShellCommand get version by running a shell command.
+func getVersionWithShellCommand(t testing.TestingT, params CheckVersionParams) (string, error) {
+	// Set appropriate binary, versionArg and extract version function
+	// based on the given parameters.
+	var binaryName, versionArg = "", DefaultVersionArg
+	switch params.binary {
+	case Docker:
+		binaryName = "docker"
+	case Packer:
+		binaryName = "packer"
+	case Terraform:
+		binaryName = "terraform"
+	default:
+		t.Fatalf("unsupported binary for checking versions.")
+	}
+
+	// Run a shell command to get the version string.
+	output, err := shell.RunCommandAndGetOutputE(t, shell.Command{
+		Command:    binaryName,
+		Args:       []string{versionArg},
+		WorkingDir: params.workingDir,
+		Env:        map[string]string{},
+	})
+	if err != nil {
+		return "", fmt.Errorf("failed to run shell command for binaray {%s} "+
+			"w/ version args {%s}: %w", binaryName, versionArg, err)
+	}
+
+	version, err := extractVersionFromShellCommandOutput(output)
+	if err != nil {
+		return "", fmt.Errorf("failed to extract version from shell "+
+			"command output {%s}: %w", output, err)
+	}
+
+	return version, nil
+}
+
+// extractVersionFromShellCommandOutput extracts version with regex string matching
+// from the given shell command output string.
+func extractVersionFromShellCommandOutput(output string) (string, error) {
+	regexMatcher, err := regexp.Compile(VersionRegexMatcher)
+	if err != nil {
+		return "", fmt.Errorf("unexpected regex compilation error: %w", err)
+	}
+
+	version := regexMatcher.FindString(output)
+	if len(version) == 0 {
+		return "", fmt.Errorf("failed to find version using regex matcher")
+	}
+
+	return version, nil
+}
+
+// validateBinaryVersionGreaterOrEqual asserts that the first version is greater
+// than or equal to the second version.
+//
+//    validateBinaryVersionGreaterOrEqual(t, 1.0.31, 1.0.27) - no error
+//    validateBinaryVersionGreaterOrEqual(t, 1.0.10, 1.0.27) - error
+//    validateBinaryVersionGreaterOrEqual(t, 1.0, 1.0.10) - error
+func validateBinaryVersionGreaterOrEqual(versionStr1 string, versionStr2 string) error {
+	if len(versionStr1) == 0 {
+		return fmt.Errorf("empty versionStr1 found")
+	} else if len(versionStr2) == 0 {
+		return fmt.Errorf("empty versionStr2 found")
+	}
```

Comment:
Super NIT: It's probably nothing more than preference, but suggesting these two ways:
```suggestion
	if len(versionStr1) == 0 || len(versionStr2) == 0 {
		return fmt.Errorf("empty versions found")
	}
```
Or
```suggestion
	if versionStr1 == "" || versionStr2 == "" {
		return fmt.Errorf("empty versions found")
	}
```

Reply:
Changed a bit so error messages are a bit different to indicate which argument is causing error.